### DISCUSSION TITLE
Wasmer 2 `Module::deserialize` error: `corrupted binary: misaligned metadata`

### DISCUSSION THREAD

**Discussion by outkine** [wasmerio/wasmer#4521] [Q&A] [answered]
Hello, I am using Wasmer and have some code that calls `wasmer::Module::deserialize` on bytes that I obtain in the following way:
```
let artifact = wasmer::UniversalArtifact::new(engine, &wasm_source, tunables).unwrap();
let bytes = artifact.serialize().unwrap();
```
I recently upgraded from Wasmer 1 to 2, and after updating the API calls `deserialize` started to panic with `corrupted binary: misaligned metadata`. 

Something strange that I noticed is that a well-crafted `println!()` invocation before and after `deserialize` makes the problem go away. However, if this works for printing strings X and Y, it does not work when there is even the slightest change to X or Y.

Does anyone have any clues as to why this may happen? Thank you! As a quick note, someone seems to be having a similar problem in [this issue](https://github.com/wasmerio/wasmer/issues/2827), but unfortunately the suggested fix of upgrading to Wasmer 3 is not an option for me.

**Comment by xdoardo** [comment, marked as answer]
Wasmer 2, 3, and 4 are not maintained anymore: is there any way we can help make the latest version an option for you to use?
